src/profile/contact-info/ContactInfoEdit.js

```diff
@@ -0,0 +1,43 @@
+import React, { useContext } from 'react';
+import { useMutation } from '@apollo/client';
+import { UPDATE_CONTACT_INFO, CREATE_CONTACT_INFO } from '../../queries/ContactInfoQueries';
+import { UPDATE_USER } from '../../queries/UserQueries';
+import { UserProfileContext } from '../UserProfileProvider';
+import { userProfileAction } from '../useUserProfile';
+import ContactInfoForm from './ContactInfoForm';
+
+function ContactInfoEdit() {
+  const [updateContactInfoCall] = useMutation(UPDATE_CONTACT_INFO);
+  const [createContactInfoCall] = useMutation(CREATE_CONTACT_INFO);
+  const [updateUserCall] = useMutation(UPDATE_USER);
+  const { state, dispatch } = useContext(UserProfileContext);
+
+  const onCancel = React.useCallback(() => {
+    dispatch({ type: userProfileAction.CANCEL, payload: 'contactInfoEdit' });
+  }, []);
+
+  const onSave = React.useCallback(async (contactInfo) => {
+    const variables = { ...contactInfo, countryId: contactInfo.country?.id };
```

Comment:
considering that `contactInfo` might be undefined, make sure you set a default value for it

Reply:
You mean a default value like a simple empty object `{}` or a predefined set of values for each field of contact info (something like `{ email: 'email', phone: '1234', ... }`) ?